cishouseholds/derive.py

```diff
@@ -1,13 +1,106 @@
 import re
 from itertools import chain
 from typing import List
+from typing import Union
 
 from pyspark.ml.feature import Bucketizer
 from pyspark.sql import DataFrame
 from pyspark.sql import functions as F
 from pyspark.sql import Window
 
 
+def assign_column_given_proportion(
+    df: DataFrame,
+    column_name_to_assign: str,
+    groupby_column: str,
+    reference_columns: List[str],
+    count_if: List[Union[str, int]],
+):
+    """
+    Assign a column boolean 1, 0 when the proportion of values meeting a condition is above 0.3
+    """
+    window = Window.partitionBy(groupby_column)
+    df = assign_true_if_any(
+        df=df, column_name_to_assign="TEMP", reference_columns=reference_columns, true_false_values=[1, 0]
+    )
+    df = df.withColumn(
+        column_name_to_assign,
+        F.when(
+            F.sum(F.when(F.col("TEMP").isin(count_if), F.lit(1)).otherwise(F.lit(0))).over(window)
+            / F.sum(F.when(F.col("TEMP").isNotNull(), F.lit(1)).otherwise(0)).over(window)
+            >= 0.3,
+            1,
+        ).otherwise(0),
+    )
+    return df.drop("TEMP")
+
+
+def count_true_row_wise(
```

Comment:
Please rename the function to reflect that it counts occurrences of a specific value in a subset of columns

Reply:
renamed